src/frontend/src/binder/expr/function.rs

```diff
@@ -343,17 +343,22 @@ impl Binder {
         }
 
         fn now() -> Handle {
-            Box::new(move |binder, mut inputs| {
-                binder.ensure_now_function_allowed()?;
-                // `now()` in batch query will be convert to the binder time.
-                if binder.is_for_batch() {
-                    inputs.push(ExprImpl::from(Literal::new(
-                        Some(ScalarImpl::Int64((binder.bind_timestamp_ms * 1000) as i64)),
-                        DataType::Timestamptz,
-                    )));
-                }
-                raw_call(ExprType::Now)(binder, inputs)
-            })
+            guard_by_len(
+                0,
+                raw(move |binder, _inputs| {
+                    binder.ensure_now_function_allowed()?;
+                    // `now()` in batch query will be convert to the binder time.
+                    if binder.is_for_batch() {
```

Comment:
some offline discussion: we can remove this and use ExprImpl::Now for batch to forbid split batch and stream logic in the binder.